octopoes/octopoes/core/service.py

```diff
@@ -128,30 +128,49 @@ def save_origin(self, origin: Origin, oois: List[OOI], valid_time: datetime) ->
         self.origin_repository.save(origin, valid_time=valid_time)
 
-    def _run_inference(self, origin: Origin, valid_time: datetime):
+    def _run_inference(self, origin: Origin, valid_time: datetime) -> None:
         bit_definition = get_bit_definitions()[origin.method]
 
-        resulting_oois = []
+        is_disabled = bit_definition.id in settings.bits_disabled or (
+            not bit_definition.default_enabled and bit_definition.id not in settings.bits_enabled
+        )
+
+        if is_disabled:
+            self.save_origin(origin, [], valid_time)
+            return
+
+        try:
+            level = self.scan_profile_repository.get(origin.source, valid_time).level
+        except ObjectNotFoundException:
+            level = 0
+
+        if level < bit_definition.min_scan_level:
+            self.save_origin(origin, [], valid_time)
+            return
 
-        if bit_definition.id not in settings.bits_disabled and (
-            bit_definition.id in settings.bits_enabled or bit_definition.default_enabled
-        ):
+        source = self.ooi_repository.get(origin.source, valid_time)
+
+        parameters_references = self.origin_parameter_repository.list_by_origin({origin.id}, valid_time)
+        parameters = self.ooi_repository.get_bulk({x.reference for x in parameters_references}, valid_time)
+
+        if bit_definition.config_ooi_relation_path is None:
             try:
-                level = self.scan_profile_repository.get(origin.source, valid_time).level
-            except ObjectNotFoundException:
-                level = 0
+                resulting_oois = BitRunner(bit_definition).run(source, list(parameters.values()))
+            except Exception as e:
+                logger.exception("Error running inference", exc_info=e)
+                return
 
-            if level >= bit_definition.min_scan_level:
-                source = self.ooi_repository.get(origin.source, valid_time)
+            self.save_origin(origin, resulting_oois, valid_time)
+            return
 
-                parameters_references = self.origin_parameter_repository.list_by_origin({origin.id}, valid_time)
-                parameters = self.ooi_repository.get_bulk({x.reference for x in parameters_references}, valid_time)
+        configs = self.ooi_repository.get_bit_configs(source, bit_definition, valid_time)
+        config = "" if len(configs) == 0 else configs[-1].config
 
-                try:
-                    resulting_oois = BitRunner(bit_definition).run(source, list(parameters.values()))
-                except Exception as e:
-                    logger.exception("Error running inference", exc_info=e)
-                    return
+        try:
+            resulting_oois = BitRunner(bit_definition).run(source, list(parameters.values()), config=config)
+        except Exception as e:
+            logger.exception("Error running inference", exc_info=e)
+            return
```

Comment:
I think we should just change the definition of every normalizer run so that it accepts a config as a third argument. If there is no config_ooi_relation_path or there is no config we can just pass an empty dictionary.